server/gc_service.go

```diff
@@ -0,0 +1,209 @@
+package server
+
+import (
+	"context"
+	"encoding/json"
+	"math"
+	"path"
+
+	"github.com/pingcap/kvproto/pkg/pdpb"
+	"github.com/pingcap/log"
+	"github.com/tikv/pd/pkg/storage/endpoint"
+	"github.com/tikv/pd/pkg/tso"
+	"github.com/tikv/pd/pkg/utils/tsoutil"
+	"go.etcd.io/etcd/clientv3"
+	"go.uber.org/zap"
+	"google.golang.org/grpc"
+)
+
+// GetGCSafePointV2 return gc safe point for the given keyspace.
+func (s *GrpcServer) GetGCSafePointV2(ctx context.Context, request *pdpb.GetGCSafePointV2Request) (*pdpb.GetGCSafePointV2Response, error) {
+	fn := func(ctx context.Context, client *grpc.ClientConn) (interface{}, error) {
+		return pdpb.NewPDClient(client).GetGCSafePointV2(ctx, request)
+	}
+	if rsp, err := s.unaryMiddleware(ctx, request, fn); err != nil {
+		return nil, err
+	} else if rsp != nil {
+		return rsp.(*pdpb.GetGCSafePointV2Response), err
+	}
+
+	rc := s.GetRaftCluster()
+	if rc == nil {
+		return &pdpb.GetGCSafePointV2Response{Header: s.notBootstrappedHeader()}, nil
+	}
```

Comment:
Do we need this check?

Reply:
I saw that other functions in GrpcServer also added this judgment.
Such as:https://github.com/tikv/pd/blob/b2275f669ff1c542b1cd0dc014c0217c1d8c5df7/server/grpc_service.go#L1462

Reply:
Yes. But here I think we don't need to check it since we only do etcd operations.

Reply:
Removed it.